tests/test_plugin.py

```diff
@@ -1534,3 +1537,83 @@ def test_3847_repro(node_factory, bitcoind):
     # This call to paystatus would fail if the pay plugin crashed (it's
     # provided by the plugin)
     l1.rpc.paystatus(i1)
+
+
+def test_important_plugin(node_factory):
+    # Cache it here.
+    pluginsdir = os.path.join(os.path.dirname(__file__), "plugins")
+
+    # Check we fail if we cannot find the important plugin.
+    n = node_factory.get_node(options={"important-plugin": "nonexistent"},
+                              may_fail=True, expect_fail=True,
+                              allow_broken_log=True)
+    assert not n.daemon.running
+    assert n.daemon.is_in_stderr('An --important-plugin=nonexistent was not found')
+
+    # Check we exit if the important plugin dies.
+    n = node_factory.get_node(options={"important-plugin": "fail_by_itself.py",
```

Comment:
it seems like you should be able to specify the plugin as either `important-plugin` or `plugin`, so here you'd only need to do

```
"important-plugin": "os.path.join(pluginsdir, "fail_by_iteself.py")
```

This does break the 'important-' / 'unimportant-' symmetry though hmm.